assets/js/util/async/index.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Finds the highest priority task from a collection of tasks with async checks.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object[]} tasks List of task objects.
```

Comment:
Let's probably mention that each object should have the priority property.

Reply:
And `check` function as well 😄